### PR TITLE
Make libs go to github when tags are built.

### PR DESCRIPTION
**Pull request by skeet70**
`skip_cleanup` may not be necessary since we're building the release as part of the deploy step. It will likely speed up that release build though, even if it isn't needed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,32 +1,50 @@
...
   only:
     - master
+
 before_install:
   - openssl aes-256-cbc -K $encrypted_487cf28eaba8_key -iv $encrypted_487cf28eaba8_iv
     -in tests/src/test/resources/service-keys.conf.enc -out tests/src/test/resources/service-keys.conf -d
 install:
   - "~/bin/sbt --help || curl -Ls https://git.io/sbt > ~/bin/sbt && chmod 0755 ~/bin/sbt"
+
+# run tests
```
Comment (coltfred) [inline .travis.yml]: Shouldn't this go on the `script:` not `before_script`.

File: .travis.yml [hunk 2]
```diff
@@ -1,32 +1,50 @@
...
     - "$HOME/.sbt/boot/"
     - "$HOME/bin"
-# hopefully prevent the cache from becoming huge - https://levans.fr/rust_travis_cache.html
-before_cache:
-  - rm -rf /home/travis/.cargo/registry
-rust:
-  - 1.34.1
+
+# build matrix
+rust: 1.34.1
```
Comment (coltfred) [inline .travis.yml]: This can be `1.35.0` now.

File: .travis_scripts/build-release.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -e
+set -x
+
+# JAVA_HOME isn't set on OSX for some reason, so manually set it
+if [ -z "$JAVA_HOME" ]; then
```
Comment (coltfred) [inline .travis_scripts/build-release.sh]: Can we do this at the top level so it doesn't have to be duplicated in both shell scripts?

File: .travis_scripts/build-release.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    export JAVA_HOME=$(/usr/libexec/java_home)
+fi
+
+cargo b --release
```
Comment (coltfred) [inline .travis_scripts/build-release.sh]: I don't think this will actually do what you want. This will do a build for release but leave the `--features ironoxide/senv` set. `cargo build --release` should be what's used here.

File: .travis_scripts/build-release.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+# copy out just the library to upload
+mkdir release_artifacts
+shopt -s extglob
```
Comment (coltfred) [inline .travis_scripts/build-release.sh]: If you wouldn't mind commenting about why this is needed, that'd be great.

File: .travis.yml (hunk 1 above)
Comment (skeet70) [reply .travis.yml]: Dunno, that's where it was before.

Comment (coltfred) [reply .travis.yml]: "run tests" wasn't in the file before... https://github.com/IronCoreLabs/ironoxide-java/blob/master/.travis.yml

File: .travis_scripts/build-release.sh (hunk 5 above)
Comment (skeet70) [reply .travis_scripts/build-release.sh]: In the deploy we need to grab the `.so` or the `.dylib`, whichever has been built for the current os. I don't want to grab all the files or all the top level files in `release` because there's junk, and globbing to `ironoxide_java.*` would get the `.d` file too. None of the commands can fail (so I can't do `release/libironoxide_java.{so,dylib}` or `-e` will bail. Without globbing (just using multiple files listed in the deploy) it is all inclusive, so it'll fail if only one is there. I looked into using `cargo --out-dir` to get only the libs without intermediate files but it's not in stable.

Basically after trying a lot of junk the solution was to use an `|` glob so if either of them is there (or both for some reason) it'll grab them and only them and put them into a spot that I can grab with the `file` directive on the `deploy` stage. `|` globs are a bash option that needs to be enabled. Hence `shopt -s extglob`. If you want that in the file as a comment, how much of it should be there?

File: .travis_scripts/build-release.sh (hunk 3 above)
Comment (skeet70) [reply .travis_scripts/build-release.sh]: Don't think you can do if-y statements like this when setting the cross shell env stuff in travis. I'm certainly no expert though. Doesn't feel like something worth spending the time to figure out though.

File: .travis.yml (hunk 1 above)
Comment (skeet70) [reply .travis.yml]: Oh the comment itself. The comment was more to label what the script is for. I was including the before_script in the "what this section is doing".

File: .travis_scripts/build-release.sh (hunk 5 above)
Comment (coltfred) [reply .travis_scripts/build-release.sh]: I would extend the comment above to say 

> Copy out just the library to upload. In order do just grab the .dylib or .so we just use extglob. We did this instead of 2 cp lines so we could leave `-e` enabled and it would succeed as long as we have at least one of them

File: .travis.yml [hunk 6]
```diff
@@ -1,32 +1,50 @@
...
   only:
     - master
+
 before_install:
   - openssl aes-256-cbc -K $encrypted_487cf28eaba8_key -iv $encrypted_487cf28eaba8_iv
     -in tests/src/test/resources/service-keys.conf.enc -out tests/src/test/resources/service-keys.conf -d
 install:
   - "~/bin/sbt --help || curl -Ls https://git.io/sbt > ~/bin/sbt && chmod 0755 ~/bin/sbt"
+
+# tests
```
Comment (clintfred) [inline .travis.yml]: What is this `tests` comment for?

Comment (skeet70) [reply .travis.yml]: It was previously `# run tests`. Just had this discussion with Colt. When I was reading this file and thinking about it, I needed to delineate what the different pieces of the file were doing. First that was moving things around to be whitespace grouped by the task they're accomplishing. With the top level `deploy` command and the caching that was now clear. I added `build matrix` to the section of `rust` and `matrix` because it's actually the combination of the two that creates the build matrix. I added `run tests` because that's what that section was there for and `script`/`before_script` didn't tell me that. Reduced it to just `tests` because Colt called out that the before scripts line specifically that it's on doesn't run the tests.

Comment (skeet70) [reply .travis.yml]: Updated again.
